webapp/components/channel_switch_modal.jsx

```diff
@@ -79,11 +97,17 @@ export default class SwitchChannelModal extends React.Component {
 
     handleSubmit() {
         const name = this.state.text.trim();
+        const currentUser = UserStore.getCurrentUser();
         let channel = null;
 
         if (name.indexOf(Utils.localizeMessage('channel_switch_modal.dm', '(Direct Message)')) > 0) {
             const dmUsername = name.substr(0, name.indexOf(Utils.localizeMessage('channel_switch_modal.dm', '(Direct Message)')) - 1);
-            channel = ChannelStore.getByName(Utils.getDirectChannelNameByUsername(dmUsername, UserStore.getCurrentUser().username).trim());
```

Comment:
`getDurectCahnnelNameByUsername` doesn't appear to be called any more so it should be removed.